ipset_test.go

```diff
@@ -236,6 +237,77 @@ func TestIPSetDifference(t *testing.T) {
 	assert.False(t, set.ContainsNet(cidr))
 }
 
+func TestIPSetIntersection(t *testing.T) {
+	testCases := []struct {
+		case1        []string
+		case2        []string
+		intersection []string
+	}{
+		{
+			[]string{"10.0.20.0/20", "10.5.8.0/24", "10.23.224.0/23"},
+			[]string{"10.0.20.0/30", "10.5.8.0/29", "10.23.224.0/27"},
+			[]string{"10.23.224.0/27", "10.0.20.0/30", "10.5.8.0/29"},
+		},
+		{
+			[]string{"10.10.0.0/30", "10.5.8.0/29", "10.23.224.0/27"},
+			[]string{"10.10.0.0/20", "10.5.8.0/24", "10.23.224.0/23"},
+			[]string{"10.10.0.0/30", "10.5.8.0/29", "10.23.224.0/27"},
+		},
+		{
+			[]string{"10.0.5.0/24", "10.5.8.0/29", "10.23.224.0/27"},
+			[]string{"10.6.0.0/24", "10.9.9.0/29", "10.23.6.0/23"},
+			[]string{},
+		},
+		{
+			[]string{"10.23.6.0/24", "10.5.8.0/29", "10.23.224.0/27"},
+			[]string{"10.6.0.0/24", "10.9.9.0/29", "10.23.6.0/29"},
+			[]string{"10.23.6.0/29"},
+		},
+		{
+			[]string{"2001:db8:0:23::/96", "2001:db8:0:20::/96", "2001:db8:0:15::/96"},
+			[]string{"2001:db8:0:23::/64", "2001:db8:0:20::/64", "2001:db8:0:15::/64"},
+			[]string{"2001:db8:0:23::/96", "2001:db8:0:20::/96", "2001:db8:0:15::/96"},
+		},
+		{
+			[]string{"2001:db8:0:23::/64", "2001:db8:0:20::/64", "2001:db8:0:15::/64"},
+			[]string{"2001:db8:0:23::/96", "2001:db8:0:20::/96", "2001:db8:0:15::/96"},
+			[]string{"2001:db8:0:15::/96", "2001:db8:0:20::/96", "2001:db8:0:23::/96"},
+		},
+		{
+			[]string{"2001:db8:0:23::/64", "2001:db8:0:20::/64", "2001:db8:0:15::/64"},
+			[]string{"2001:db8:0:14::/96", "2001:db8:0:10::/96", "2001:db8:0:8::/96"},
+			[]string{},
+		},
+		{
+			[]string{"2001:db8:0:23::/64", "2001:db8:0:20::/64", "172.16.1.0/24"},
+			[]string{"2001:db9:0:14::/96", "2001:db9:0:10::/96", "172.16.1.0/28"},
+			[]string{"172.16.1.0/28"},
+		},
+	}
+	for i, testCase := range testCases {
```

Comment:
Could you break this up into individual test cases? This for loop over the above list of test cases is a pattern that makes it just a little cumbersome to map failures back to the problem. You can take L288 - L307 and make a utility function out of it. Then you can make the test cases above into separate Test functions that each call the utility.

The advantage is that failures give a stack trace that makes it clear which case has failed. Make sense?